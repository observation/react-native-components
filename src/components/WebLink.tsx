import React from 'react'
import { StyleProp, TextStyle, ViewStyle } from 'react-native'

import { IconExternalLink } from './Icon'
import IconText from './IconText'
import textStyles from '../styles/text'

type Props = {
  onPress: () => void
  containerStyle?: StyleProp<ViewStyle>
  textStyle?: StyleProp<TextStyle>
  text: string
}

const WebLink = ({ onPress, containerStyle, text, textStyle }: Props) => (
  <IconText
    icon={<IconExternalLink />}
    text={text}
    style={{
      containerStyle,
      textStyle: [textStyles.link, textStyle],
    }}
    onPress={onPress}
  />
)

export default WebLink
