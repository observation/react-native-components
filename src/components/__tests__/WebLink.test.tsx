import React from 'react'

import { fireEvent, render } from '@testing-library/react-native'

import WebLink from '../WebLink'

const onPress = jest.fn()
describe('WebLink', () => {
  test('Rendering', () => {
    const { toJSON } = render(<WebLink onPress={onPress} containerStyle={{ flex: 1 }} text="View on website" />)
    expect(toJSON()).toMatchSnapshot()
  })

  test('Click', async () => {
    const { getByText } = render(<WebLink onPress={onPress} containerStyle={{ flex: 1 }} text="View on website" />)
    await fireEvent.press(getByText('View on website'))
    expect(onPress).toBeCalled()
  })
})
