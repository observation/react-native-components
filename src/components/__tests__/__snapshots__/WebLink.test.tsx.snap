// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebLink Rendering 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "justifyContent": "center",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      <Icon
        color="#0066B1"
        name="arrow-up-right-from-square"
        size={14}
        type="light"
      />
    </View>
    <Text
      style={
        [
          {
            "flexShrink": 1,
          },
          [
            {
              "color": "#0066B1",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
          ],
        ]
      }
    >
      View on website
    </Text>
  </View>
</View>
`;
